pkg/controller/kieapp/defaults/defaults.go

```diff
@@ -31,25 +31,32 @@ func GetEnvironment(cr *v1.KieApp, client client.Client) (v1.Environment, error)
 	var common v1.Environment
 	yamlBytes, err := loadYaml(client, "common.yaml", cr.Namespace, envTemplate)
 	if err != nil {
+		logrus.Error(err)
```

Comment:
Generally speaking, a function returning an error shouldn't need to log it. It's letting the caller know that it failed, and that's enough. The caller would then either bubble up the error, or handle it. The final caller up the stack handling the error would log it. This would avoid multiple redundant error messages.

Another reason for this is that only the caller knows what's an error and what's not. Imagine that a unit test somehow removed or corrupted common.yaml, expecting this function to return an error. It would then declare success at seeing the error and logging something at the error verbosity would not be appropriate. This may not make much sense for this specific instance, but that's a good reason to do the above.

Reply:
agree

Reply:
done